src/controllers/Interactions.ts

```diff
@@ -0,0 +1,38 @@
+import { Interaction, Message, TextChannel } from "discord.js";
+import QueueRepository from "../repositories/QueueRepository";
+import { joinQueue, leaveQueue } from "../services/QueueService";
+import MessageBuilder, { ButtonCustomID } from "../utils/MessageBuilder";
+
+export async function postCurrentQueue(queueChannel: TextChannel): Promise<void> {
+  const ballchasers = await QueueRepository.getAllBallChasersInQueue();
+  await queueChannel.send(MessageBuilder.queueMessage(ballchasers));
+}
+
+export async function handleInteraction(buttonInteraction: Interaction): Promise<void> {
+  if (!buttonInteraction.isButton()) return;
+
+  const message = buttonInteraction.message;
+  if (!(message instanceof Message)) return;
+
+  await buttonInteraction.deferUpdate();
+
+  switch (buttonInteraction.customId) {
+    case ButtonCustomID.JoinQueue: {
+      await message.edit(MessageBuilder.disabledButtonsJoining());
+      const ballchasers = await joinQueue(buttonInteraction.user.id, buttonInteraction.user.username);
+      await message.edit(MessageBuilder.queueMessage(ballchasers));
+      break;
+    }
+    case ButtonCustomID.LeaveQueue: {
+      await message.edit(MessageBuilder.disabledButtonsLeaving());
+      const remainingMembers = await leaveQueue(buttonInteraction.user.id);
+
+      if (remainingMembers) {
```

Comment:
Do you need to do this if check. I think MessageBuilder handles this check for you.

Reply:
QueueService.ts handles the check, but it returns PlayerInQueue or null. This is why there is a check to see what we were returned here.

Reply:
`if (remainingMembers) {
        await message.edit(MessageBuilder.queueMessage(remainingMembers));
      } else {
        await message.edit(MessageBuilder.queueMessage([]));
      }`
I changed the code to resend the queueMessage with an empty array if it returns null. If there is no else case, then Norm sits idle and wont update the buttons from being disabled.

Reply:
You could remove the if else all together I believe.  If you change MessageBuilder to also accept null 
![image](https://user-images.githubusercontent.com/14949544/155224126-0ea9662d-4bb0-4920-abad-1ac62d26b5f3.png)
You could then just send remainingMembers to queueMessage().

Reply:
You can do @duanekord 's suggestion, or edit `leaveQueue` to return an empty array instead of null. Personally, I think an empty array would be a bit cleaner, but it doesn't matter too much.

Reply:
I changed leaveQueue to not return null and instead just the empty array. It no longer returns null at all. The check is removed from the Case in Interactions.ts. Changed the integration test to accept `[]` instead of null.